assets/test/socket_test.js

```diff
@@ -270,6 +270,29 @@ describe("with transports", done =>{
       assert.equal(socket.conn, null)
     })
 
+    it("triggers onClose callback", done => {
+      const spy = sinon.spy()
+      socket.onClose(spy)
+
+      socket.connect()
+
+      socket.disconnect(() => {
+        assert(spy.calledOnce)
+        done()
+      });
+    });
+
+    it("sets channels to error state", () => {
+      socket.connect()
+
+      const channel = socket.channel("topic", {one: "two"})
+      channel.join().trigger("ok", {})
+      assert.equal(channel.state, "joined")
+
+      socket.disconnect()
+      assert.equal(channel.state, "errored")
```

Comment:
I'm trying to bridge the gap between what might happen in tests and what happens in prod.  Basically, I am assuming `mock-socket` sets the `readyState` to CLOSED and then follows the following steps - [spec](https://html.spec.whatwg.org/multipage/web-sockets.html#feedback-from-the-protocol%3Aconcept-websocket-closed) - which is to fire an `error` event, thus setting the `state === 'errored'` (our code) and firing a `close` event...all happening in a [task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-a-task).  

Example tests asserting the `errored` state when calling `socket.disconnect()` w/ `setTimeout`:
https://github.com/snewcomer/phoenix/commit/fef70317cc41f445d92371475543b126b896e914

So, overall `onConnClose` is called from what I can see.  Just in the next macroTask.

Reply:
Ok just tested a running app an do not see `onConnClose` called.  So in a test context, this test ensures you don't need to `assert` at some point in the future b/c you made the `onConnClose` sync instead of async.

Thinking in my own head, I wonder how we could expand this test given the diff b/c test and prod...